src/preprocessing/parse_data.py

```diff
@@ -0,0 +1,160 @@
+import xml.etree.ElementTree as ET
+import json, re
+from pathlib import Path
+import os
+
+
+class PlenarprotokollXMLParser:
+    def __init__(self):
+        self.data = dict()
+
+    # due to the xml-files containing invisible and ambiguous characters they have to be removed
+    def remove_bad_chars(self, text: str) -> str:
+        invisible_spaces_pattern = re.compile(r"[\u00A0]")
+        invisible_pattern = re.compile(r"[\u202F]")
+        bad_dashes_pattern = re.compile(r"[\u2013]")
+        bad_quotation_pattern = re.compile(r"[\u201c\u201e]")
+
+        cleaned_text = invisible_spaces_pattern.sub(" ", text)
+        cleaned_text = invisible_pattern.sub("", cleaned_text)
+        cleaned_text = bad_dashes_pattern.sub("-", cleaned_text)
+        cleaned_text = bad_quotation_pattern.sub("'", cleaned_text)
+
+        return cleaned_text
+
+    def extract_spoken_comments(self, comment: str) -> list[dict]:
+        pattern = re.compile(
+            r"([A-Za-zäöüÄÖÜßğ\.\s]+) \[([A-Za-z0-9äöüÄÖÜß\s/]+)\]: (.+)"
+        )
+        segments = comment.split("-")
+        extracted_info = []
+        for segment in segments:
+            matches = pattern.findall(segment)
+            for match in matches:
+                person, party, text = match
+                if text.endswith(")"):
+                    text = text[:-1].strip()
```

Comment:
Nur ein Vorschlag und nitpicking:
```suggestion
                text = text.rstrip(")")
```